### PR TITLE
[CLOUDSTACK-9337]Enhance vcenter.py to create data center in vcenter server automatically

### PR DESCRIPTION
**Pull request by sanju1010**
These changes have been made to support vmware deployments in CI. 
For CI to create cloudstack setup with vmware, it is required to create datacenter, cluster and hosts in vcenter server before adding in cloudstack. Added few methods in vcenter.py to perform these tasks.

Please refer to CLOUDSTACK-9337 for more details.

### CODE REVIEW

File: tools/marvin/marvin/lib/vcenter.py [hunk 1]
```diff
@@ -183,7 +192,157 @@ def get_clusters(self, dc, clus=None):
...
+        """
+        if len(dcname) > 79:
+            raise ValueError("The name of the datacenter must be under "
+                             "80 characters.")
+
+        if folder is None:
+            folder = self.service_instance.content.rootFolder
+
+        if folder is not None and isinstance(folder, vim.Folder):
+            dc_moref = folder.CreateDatacenter(name=dcname)
```
Comment (alexandrelimassantana) [inline tools/marvin/marvin/lib/vcenter.py]: Hello @sanju1010 
Is it necessary to double check if folder is not None here? Also you mentioned some restrictions for the dcname, shouldnt they also raise and ValueError?

Comment (sanju1010) [inline tools/marvin/marvin/lib/vcenter.py]: @alexandrelimassantana , CreateDatacenter will through appropriate exception if the dcname is not proper. Double check for folder is to make sure that we get the rootFolder object if folder is None.

**Comment by shwetaag** [comment]
Did a code walk through . LGTM . 
Get the CI result for this PR

File: tools/marvin/marvin/lib/vcenter.py (hunk 1 above)
Comment (alexandrelimassantana) [inline tools/marvin/marvin/lib/vcenter.py]: got it. The code looks good, those were the only doubts I had. I would just use a Semaphore in line 317, but sleep an pool is not that bad also.

The code seems fine, but tests are needed.

**Comment by swill** [comment]
I don't have the ability to test vcenter.  Is there anyone else who has the ability to test vcenter who can post the test results of a CI run?

**Comment by sanju1010** [comment]
These changes will not have any impact on cloudstack code since it is purely to orchestrate the vcenter Server to create datacenter->cluster->hosts. Can we merge this without waiting for CI to run?

**Comment by koushik-das** [comment]
@shwetaag @swill As I understand from @sanju1010 comment, the script is used to automate creation of dc, cluster and hosts in vCenter. The setup created can then be used to run Cloudstack tests for VMware. I don't see the need to run CI on this.

**Comment by swill** [comment]
I am fine with not running CI on this one.  It would be nice if someone could post a screenshot or something to show that the code behaves as expected.

We are missing 1 LGTM, otherwise I think we are in pretty good shape.

**Comment by rhtyd** [comment]
@sanju1010 how can we test this

tag:vmware-pickup

**Comment by swill** [comment]
anyone able to pick this up for validation and giving me a final code review???

**Comment by swill** [comment]
This one is pretty much ready, but without verification that it works and without the required code reviews I can't really justify merging it.  I just don't know if works and does not break anything...  Not sure what to do with this one...  4.9 freeze is VERY close right now, so time is of the essence...

**Comment by swill** [comment]
@sanju1010 I don't know what to do with this one.  I don't have any verification that this PR works, so I am not sure what information I am supposed to be going on to be confident in merging it.

**Comment by bvbharatk** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 165
 Hypervisor xenserver
 NetworkType Advanced
 Passed=73
 Failed=0
 Skipped=3

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**

**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vpc_vpn.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_routers.py
test_reset_vm_on_reboot.py
test_snapshots.py
test_deploy_vms_with_varied_deploymentplanners.py
test_login.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by sanju1010** [comment]
@swill , BVT results are clean. Can we push this to master? This will not have any impact on acs master stablity.

**Comment by swill** [comment]
Master is frozen in prep for the 4.9 release. This will have to wait for 4.10.

**Comment by karuturi** [comment]
👍 merging this now.

**Pull request merged**